### PR TITLE
FSDE-24709-Add dropdown web component documentation

### PR DESCRIPTION
**Pull request by soujanreddy**

### CODE REVIEW

File: src/custom-elements/src/components/button/button.tsx [hunk 1]
```diff
@@ -18,6 +18,11 @@ export class Button {
    */
   @Prop({ reflect: true }) variant: string;
 
+  /**
+   *  to set data position of a button { outline, flat }.
+   */
+   @Prop({ reflect: true }) dataPosition?: string;
```
Comment (jllr) [inline src/custom-elements/src/components/button/button.tsx:24]: @soujanreddy for what is needed this prop?

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -35,6 +36,10 @@ export class Dropdown {
    * To provide the value of base-style button as trigger of the Dropdown
    */
   @Prop() button?: string;
+  /**
+   * To animate the accordion of Dropdown
+   */
+  @Prop() animate?: boolean;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:42]: @soujanreddy this prop is making the PR failing when build. One option would be to change the name to `iconAnimation` or something along these lines

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 3]
```diff
@@ -43,6 +48,10 @@ export class Dropdown {
    * To provide selector of an external reference element
    */
   @Prop() reference: string;
+  /**
+   * To provide style for dropdown menu
+   */
+  @Prop() menuStyle?: string;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:54]: @soujanreddy same here for this prop.

File: src/website/views/components/dropdown/_properties.pug [hunk 4]
```diff
@@ -1,7 +1,151 @@
 h2 JavaScript
-h3 Properties
+h3.-mt--2 Properties
```
Comment (jllr) [inline src/website/views/components/dropdown/_properties.pug:2]: @soujanreddy Web component Properties are compiled automatically adding `:stencilDoc(name='chi-dropdown')`

**Review by jllr — CHANGES_REQUESTED** [review]

File: src/custom-elements/src/components/button/button.tsx (hunk 1 above)
Comment (soujanreddy) [reply src/custom-elements/src/components/button/button.tsx:24]: @jllr this is to position the dropdown menu for the dropdown button. The current dropdown component is not passing the positions to chi-button.

File: src/custom-elements/src/components/dropdown/dropdown.tsx (hunk 3 above)
Comment (soujanreddy) [reply src/custom-elements/src/components/dropdown/dropdown.tsx:54]: @jllr You mean to update the name of this prop?

File: src/website/views/components/dropdown/_properties.pug (hunk 4 above)
Comment (soujanreddy) [reply src/website/views/components/dropdown/_properties.pug:2]: I see we don't have a data object set for chi-dropdown in docs.json. I might have to add that object first to use :stencilDoc(name='chi-dropdown')

File: src/custom-elements/docs/docs.json [hunk 5]
```diff
@@ -1612,6 +1612,210 @@
...
+            {
+              "type": "boolean"
+            }
+          ],
+          "optional": false,
+          "required": false
+        }
+      ],
+      "methods": [],
+      "events": [],
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/docs/docs.json:1744]: You can also document methods and events.

File: src/custom-elements/docs/docs.json [hunk 6]
```diff
@@ -1612,6 +1612,153 @@
...
+      "docsTags": [],
+      "usage": {},
+      "props": [
+        {
+          "name": "active",
+          "type": "boolean",
+          "mutable": false,
+          "attr": "active",
+          "reflectToAttr": false,
+          "docs": "to set the state of Dropdown.",
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/docs/docs.json:1629]: Capitalise all descriptions.

```suggestion
          "docs": "To set the state of Dropdown.",
```

File: src/website/views/components/dropdown/_properties.pug [hunk 7]
```diff
@@ -1,7 +1,8 @@
-h2 JavaScript
-h3 Properties
+h2 Web Component
+:stencilDoc(name='chi-dropdown')
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/dropdown/_properties.pug:3]: You can follow this example where we have all documented: 

https://assets.ctl.io/chi/5.16.0/components/drawer/?theme=Portal#web-component

As you can see the Web components, Vue and Javascript have their own properties, methods and events so in the dropdown we need to follow the same pattern.

The `stencilDoc` will automatically add the properties, methods and events for the web components, but we still need the Javascript part.

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]
We cannot see the description in the methods because the description has to be added as a comment in the `dropdown.tsx` file as in `drawer.tsx` and then generate the docs.json again.

dropdown.tsx
image

drawer.tsx
image

**Comment by soujanreddy** [comment]
> We cannot see the description in the methods because the description has to be added as a comment in the `dropdown.tsx` file as in `drawer.tsx` and then generate the docs.json again.
> 
> dropdown.tsx image
> 
> drawer.tsx image

Done, updated the comments and descriptions.

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]
I noticed that you have not documented the menu section. It is possible to create only menus with the dropdown web component.

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 8]
```diff
@@ -27,6 +28,10 @@ export class Dropdown {
    * To set the state of Dropdown
    */
   @Prop() active: boolean;
+  /**
+   * To set the description of Dropdown menu item
+   */
+   @Prop() enableDescription?: boolean;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:34]:
```suggestion
   @Prop() description?: boolean;
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 9]
```diff
@@ -212,7 +230,7 @@ export class Dropdown {
     ) : null;
     const menu = (
       <div
-        class={`${DROPDOWN_CLASSES.MENU} ${this.active ? ACTIVE_CLASS : ''}`}
+        class={`${DROPDOWN_CLASSES.MENU} ${this.active ? ACTIVE_CLASS : ''} ${this.enableDescription ? '-list' : ''}`}
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:233]:
```suggestion
        class={`${DROPDOWN_CLASSES.MENU} ${this.active ? ACTIVE_CLASS : ''} ${this.description ? '-list' : ''}`}
```

File: src/custom-elements/src/components/checkbox/checkbox.tsx [hunk 10]
```diff
@@ -20,6 +20,10 @@ export class Checkbox {
    * To provide checkbox label as a string
    */
   @Prop() label: string;
+  /**
+   * To add icon beside the checkbox
+   */
+   @Prop() iconCheckbox?: string;
```
Comment (jllr) [inline src/custom-elements/src/components/checkbox/checkbox.tsx:26]:
```suggestion
```

File: src/custom-elements/src/components/checkbox/checkbox.tsx [hunk 11]
```diff
@@ -106,6 +110,7 @@ export class Checkbox {
           type="checkbox"
         />
         <label class={`${CHECKBOX_CLASSES.LABEL} ${this.state ? `-${this.state}` : ''}`} htmlFor={`${this.id}-control`}>
+          {this.iconCheckbox && <i class={`chi-icon icon-${this.iconCheckbox}`} aria-hidden="true"></i>}
```
Comment (jllr) [inline src/custom-elements/src/components/checkbox/checkbox.tsx:113]:
```suggestion
```

**Review by jllr — CHANGES_REQUESTED** [review]

**Review by jllr — APPROVED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

**Pull request merged**